client/src/session_manager.rs

```diff
@@ -393,17 +392,19 @@ impl SessionManager {
             Err(_) => (self.server_session().await?, node.slot),
         };
 
-        Ok(self
-            .registry
-            .add_node(parse_node_id(&node.node_id)?, session.clone(), slot)
-            .await)
+        let node_id = (&node.node_id)
+            .try_into()
+            .map_err(|e| anyhow!("Error parsing NodeId: {}", e))?;
+        Ok(self.registry.add_node(node_id, session.clone(), slot).await)
     }
 
     pub async fn try_direct_session(
         &self,
         packet: &proto::response::Node,
     ) -> anyhow::Result<Arc<Session>> {
-        let node_id = parse_node_id(&packet.node_id)?;
+        let node_id = (&packet.node_id)
+            .try_into()
+            .map_err(|e: &str| anyhow::anyhow!(e))?;
```

Comment:
You are returning `&str` as an error type from `TryFrom`, that's why you need to map error here. Please return better error type. In general everything what we return as an error should be convertible to anyhow by using `?`